lib/pages/caregiver/calendar_page.dart

```diff
@@ -0,0 +1,350 @@
+
+	List<Color> markerColors = [
+		Colors.green,
+		Colors.pink,
+		Colors.deepPurple,
+		Colors.teal,
+		Colors.orange,
+		Colors.red,
+		Colors.purple,
+		Colors.brown
+	];
+
+	@override
+	void initState() {
+		super.initState();
+		_calendarController = CalendarController();
+		_animationController = AnimationController(vsync: this, duration: _animationDuration);
+		_animationController.forward();
+	}
+
+	@override
+	void dispose() {
+		_calendarController.dispose();
+		_animationController.dispose();
+		super.dispose();
+	}
+
+	@override
+	Widget build(BuildContext context) {
+		return Scaffold(
+			body: Column(
+				crossAxisAlignment: CrossAxisAlignment.start,
+				children: [
+					AppHeader.widget(
+						title: '$_pageKey.header.title',
+						appHeaderWidget: Card(
+							margin: EdgeInsets.symmetric(horizontal: 12.0).copyWith(bottom: 6.0),
+							child: InkWell(
+								onTap: () => {},
+								child: Container(
+								  child: BlocBuilder<CalendarCubit, CalendarState>(
+									  builder: (context, state) {
+										  if (state.children == null) {
+											  context.bloc<CalendarCubit>().loadInitialData();
+											  return buildChildPicker(loading: true);
+										  }
+											int markerIndex = 0;
+											_selectedChildren = [];
+											state.children.forEach((key, value) {
+												if(value)
+													_selectedChildren.add(key);
+											});
+											for(var child in state.children.keys) {
+												if(markerIndex == markerColors.length)
+													markerIndex = 0;
+												_childrenColors[child.id] = markerColors[markerIndex++];
+											}
+									    return buildChildPicker(children: state.children);
+									  },
+								  ),
+								)
+							)
+						)
+					),
+					AppSegments(
+						segments: [
+							Padding(
+								padding: EdgeInsets.symmetric(horizontal: AppBoxProperties.screenEdgePadding),
+								child: BlocBuilder<CalendarCubit, CalendarState>(
+									builder: (context, state) {
+										return buildCalendar(state.events);
+									},
+								)
+							),
+							Padding(
+								padding: EdgeInsets.only(top: AppBoxProperties.sectionPadding),
+								child: buildPlanList()
+							)
+						]
+					)
+				]
+			)
+		);
+	}
+
+	Widget buildPlanList(){
+		return BlocBuilder<CalendarCubit, CalendarState>(
+			builder: (context, state) {
+				return Segment(
+					title: '$_pageKey.content.plansOnDateTitle',
+					titleArgs: {'DATE': DateFormat.yMd(Localizations.localeOf(context).toString()).format(state.day).toString()},
+					noElementsMessage: '$_pageKey.content.noPlansOnDateTitle',
+					noElementsIcon: Icons.description,
+					elements: [
+						if(state.events != null && state.events[state.day] != null)
+							for(UIPlan plan in state.events[state.day])
+								ItemCard(
+									title: plan.name,
+									onTapped: () => Navigator.pushNamed(context, AppPage.caregiverPlanDetails.name, arguments: {'planID': plan.id}),
```

Comment:
You can just pass plan.id without the map. Unless you plan to add more fields to it later.